Add EIP-1014 (CREATE2) address test as a first runCall.js API test

jochem-brouwer opened the pull request:
Add an explicit CREATE2 spam test and verify if the expected address matches the actual address. The test passes.

Linked issue: #777 

The test in the issued failed on their end; VM all good. We might want to add this test anyways.

jochem-brouwer commented:
The issue is closed but we might want to add this test anyways. Thoughts? @holgerd77

holgerd77 commented on packages/vm/tests/api/constantinople/eip-1014.js:
```diff
@@ -0,0 +1,72 @@
+        // pad string if necessary
+        if (hexString.length % 2 != 0) {
+            hexString = "0" + hexString
+        }
```
Think you can use `padToEven` from the `ethjsUtil` methods from [ethereumjs-util](https://github.com/ethereumjs/ethereumjs-util#ethjs-util-methods) here.

holgerd77 commented on packages/vm/tests/api/constantinople/eip-1014.js:
```diff
@@ -0,0 +1,72 @@
+
+    t.end()
+})
+
```
Can you rename the filename to `eip-1014-create2.js` to make this a bit more descriptive since likely not everyone will be able to associate EIPs with the specific topic?

jochem-brouwer replied:
I will do this, but I want to note that in /tests/api/istanbul/ we also have eip-* tests without name =) I agree this is a good idea though.

holgerd77 commented:
@jochem-brouwer Thanks! Was a bit unsure here at first if to keep or not, we should also pay attention to not overblow the code base, since all code we add at the end adds some complexity and things which need to be understood by someone else at some point.

But I think in this case we might want to keep, think this can be useful to have this tested from the API level.

jochem-brouwer commented:
@holgerd77 Yep I agree. I am not super keen to add this at all but I nevertheless just created this PR as I wrote this test in order to investigate the related issue. It is a bit of a waste to just throw the test away without any discussion if it might be worth adding it 😄

holgerd77 commented:
Just seen that we generally don't have any dedicated `runCall` tests in `api/` - so as `api/runCall.js`. Since you are setting up on `runCall` here: do we want to take this test as some basis for `runCall` tests and move this over and generalize the code a bit (by extracting useful setup parts) so this can serve as a starting point? 🤔  Think this might be a good fit.

jochem-brouwer commented:
@holgerd77 I am not really sure what parts of `runCall` we want to test here? If you have some suggestions then I'd be okay creating some `runCall` generalized tests here.

holgerd77 commented:
@jochem-brouwer The API of `runCall()`, your test is already doing that for some call input to the function and then comparing parts of the result, see e.g. [runBlock.js](https://github.com/ethereumjs/ethereumjs-vm/blob/master/packages/vm/tests/api/runBlock.js) for a similar setup.

I would just suggest that we could use your test and some starting point here and move this one test over, not that you necessarily need to expand on this for now.

Just an idea, since I noticed that we have `runBlock.js`, `runCode.js`,... but no `runCall.js` test file, so let me know if you don't think your test would fit in this changed context.

jochem-brouwer commented:
@holgerd77 I agree, it makes sense to include this test and use it as a future starting point for `runCall` tests. I have changed/added your suggestions.

holgerd77 commented:
Hi @jochem-brouwer, can you directly move the test file to `runCall.js` and eventually adopt the code comments a bit to reflect more the general nature of the test file (if/where makes sense)? I think if we don't do this know this might get forgotten.

It might also be good to do a rebase along and squash the commits into one, otherwise things are getting too bloated for such a targeted and not too distributed change.

jochem-brouwer commented:
I changed the filename and added some more comments. 

I tried squashing the commits but this didn't work out so well. Now there's 11 commits instead of the previous 3. Woops. Need to step up my git magic. Can someone help me fix this? I want to squash my own commits together. I can manually do it by just `git checkout master`, creating a new branch and just dumping this single file in it (`runCall.ts`) and then committing it. But I'd rather know how I can squash these branch commits together. I think I went wrong by first rebasing from master and then trying to squashing my own commits? (Wrong order?)

jochem-brouwer commented:
Okay I fixed it using the manual method, but pretty sure I won't mess this up next time (in a bigger PR 😅 )

holgerd77 commented:
@jochem-brouwer thanks, you can do from the branch you are working on with `git rebase -i master` (-i for "interactive" mode), then you get an editor view with your commits together with some explanation what you can do. I am personally always super careful here and e.g. never do two things at once but just all in single atomic steps (by calling `git rebase -i master` a couple of times), all this rebasing stuff still makes me somewhat afraid on every new round. 😋
